src/v/cluster/topics_frontend.cc

```diff
@@ -62,6 +62,26 @@
 #include <sstream>
 #include <system_error>
 
+namespace {
+
+std::optional<std::string_view> get_enterprise_feature(
+  const cluster::custom_assignable_topic_configuration& cfg) {
+    if (
+      cfg.cfg.properties.shadow_indexing
```

Comment:
nit: is it worth asserting here that `shadow_indexing` is non-nullopt by this point? Or using `value_or` to be defensive.

Reply:
given that if `cfg.cfg.properties.shadow_indexing` is `nullopt` it will currently return `true`, i say `value_or` :+1:

Reply:
It should always be set, but I'll change it to this:
```c++
    const auto si_disabled = model::shadow_indexing_mode::disabled;
    if (
      cfg.cfg.properties.shadow_indexing.value_or(si_disabled) != si_disabled) {
```

Reply:
>if `cfg.cfg.properties.shadow_indexing` is `nullopt` it will currently return `true`

oh yeah...cursed overload lol

Reply:
Fixed